src/env_model_repository.cpp

```diff
@@ -47,6 +47,35 @@ void EnvironmentModelRepository::updateItem(const std::string& name, std::shared
   items[name]->setPayload(plptr);
 }
 
+void EnvironmentModelRepository::removeItem(const std::string& name)
+{
+  std::shared_ptr<Item> to_be_erased = items[name];
+  // Remove the reference to the "item_to_be_erased" from its parent item
+  if(!to_be_erased->isRoot())
+  {
+    auto parent_of_to_be_erased = to_be_erased->getParent();
+    std::shared_ptr<Item> parent_ptr = parent_of_to_be_erased.lock();
+    auto it = std::remove_if(parent_ptr->getChildrenNonConst()->begin()
```

Comment:
instead of invoking `parent_ptr->getChildrenNonConst()` all the time do
`std::vector<shared_ptr<Item> parents_children& = parent_ptr->getChildrenNonConst()` and use this reference